src/mainwindow.cpp

```diff
@@ -9700,6 +9707,43 @@ void MainWindow::on_noteSubFolderTreeWidget_currentItemChanged(
     reloadTagTree();
 }
 
+void MainWindow::on_noteSubFolderTreeWidget_itemSelectionChanged()
+{
+    auto items = ui->noteSubFolderTreeWidget->selectedItems();
+    //if no items selected
+    if (items.count() < 1) {
+        return;
+    }
+
+    QList<int> noteSubFolderIds;
+    QList<int> noteIdList;
+    //get all the folder ids
+    Q_FOREACH(QTreeWidgetItem *item, items) {
+        noteSubFolderIds << item->data(0, Qt::UserRole).toInt();
+    }
+
+    // get the notes from the subfolders
+    Q_FOREACH(int noteSubFolderId, noteSubFolderIds) {
+        // get all notes of a note sub folder
+        QList<Note> noteList = Note::fetchAllByNoteSubFolderId(
+                    noteSubFolderId);
+        noteIdList << Note::noteIdListFromNoteList(noteList);
+    }
+
+    QTreeWidgetItemIterator it(ui->noteTreeWidget);
+
+    while (*it) {
+        // hide all notes that are not in selection
+        if (!noteIdList.contains((*it)->data(0, Qt::UserRole).toInt())) {
+            (*it)->setHidden(true);
```

Comment:
You cannot simply show or hide notes in the list. Don't forget about everything MainWindow::filterNotes() does! :smile:

Reply:
Really? :thinking: 
But it does hide notes, and shows the ones from the selected folders :thinking: 
I don't know the implications this will have for all other modules though and the side effects...

`MainWindow::filterNotes()` would require a lot of changes to work with multi folder selection, right? or maybe I didn't understand you

Reply:
You can always for example also search for text at the same time, working with `MainWindow::filterNotes()` is mandatory or you will immediately get bug reports. :grimacing:

Reply:
Your new filters need to be part of the `MainWindow::filterNotes()` filtering process.

Reply:
> You can always for example also search for text at the same time, working with `MainWindow::filterNotes()` is mandatory or you will immediately get bug reports. grimacing

I just tried this.
Multi select + search in noteTreeWidget. Works, but if you delete the searched text, the selection isn't retained.

Reply:
> Your new filters need to be part of the `MainWindow::filterNotes()` filtering process.

Ok, let me try this.

Reply:
Yes, Now its working much better I think

Reply:
and if someone changes the search text after you selected note sub-folders? :grin:

Reply:
:grimacing: 
The notes aren't getting filtered

Reply:
Ok, so now the filtering is working with multi folder selection

Reply:
Searching for text seems to work, but if I also select a tag than no notes are shown if more than one sub-folder was selected.

Reply:
And if multiple sub-folders are selected then only the notes from the selected sub-folders are shown, not also their sub-folders (if the "including sub-folders" checkbox is checked in the settings)

Reply:
> And if multiple sub-folders are selected then only the notes from the selected sub-folders are shown, not also their sub-folders (if the "including sub-folders" checkbox is checked in the settings)

Yes, I didn't do that because I thought the user might only want to see notes from the selected folder and note the subfolders too :thinking: 
Ok then, I will enable this if 'including-subfolders' is checked

> Searching for text seems to work, but if I also select a tag then no notes are shown if more than one sub-folder was selected.

Yes. That is not done yet. This is for round 2. :grin: 

Round 2 beings now.

Reply:
Notes of subfolders are now also visible if show recursive setting is on.

Tags are working, but not as well as they should.
Tag count doesn't get updated for multiple folders.

The only way to fix that would be.. to modify `NoteSubFolder` class to take account of 'multiple-active-note-subfolders` and have another option like `isMultipleActive`

What do you think?

Reply:
> Notes of subfolders are now also visible if show recursive setting is on.

https://imgur.com/a/ESYI0w7

With `Ctrl + Click` I was able to remove a selected sub-folder but the not the notes from the still selected sub-folder was shown. (there should be shown 15 notes in the end)

Reply:
> Tags are working, but not as well as they should.

https://imgur.com/a/03QgQVq

Doesn't work for me, one sub-folder and one tag selected shows one note. when I add a 2nd sub-folder to the multi-selection no notes are shown

> Tag count doesn't get updated for multiple folders.

that would be ok for me, we can just note that limitation. of course if you also want to fix that you are free to do so, would save us the bug-report handling :laughing:

Reply:
> Doesn't work for me, one sub-folder and one tag selected shows one note. when I add a 2nd sub-folder to the multi-selection no notes are shown

Yes, that's because when you click the second folder, that folder becomes active and since the second folder has  no associated tags with 'art', no notes are shown.

Tag only fetches notes for a certain selected folder at a time, right?
 
> > Tag count doesn't get updated for multiple folders.
> 
> that would be ok for me, we can just note that limitation. of course if you also want to fix that you are free to do so, would save us the bug-report handling laughing

Reply:
Alright, tags are now working quite well I think

Reply:
> Alright, tags are now working quite well I think

clicking on tags *after* selecting multiple su-folders doesn't seem to reduce the amount of notes shown in the note list :grin:

Reply:
Ah Yes. Ok let me fix that.

Reply:
> clicking on tags after selecting multiple sub-folders doesn't seem to reduce the amount of notes shown in the note list

I still have that issue.

Reply:
Can you make a small vid like before?

Reply:
Do you maybe have the 'recursive tagging' setting on? 🤔

Reply:
![Peek-2019-11-30-12-11](https://user-images.githubusercontent.com/7696024/69897244-ebb21880-136a-11ea-86a0-33c507e8a396.gif)

Reply:
Please say something here too :grin:

Reply:
Need more time to test that again, maybe tomorrow. I'm out the whole day today. 😬

Reply:
Alright :grin:

Reply:
> Do you maybe have the 'recursive tagging' setting on? thinking

yes

https://imgur.com/a/7cr6b2a

Multiple sub-folders selected. Clicking on tags didn't change the note list.

Reply:
I am sorry, I tested one more time and didn't have the issue, I guess tray icon support shot me in the leg (an old version of the app may not have been closed). :grimacing:

Reply:
Cool. One more feature down. :1st_place_medal: :grin: